nebula/ui/components/VPNSegmentedToggle.qml

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+import Mozilla.VPN 1.0
+
+// VPNSegmentedToggle
+Rectangle {
+    id: root
+
+    property alias model: options.model
+    property int selectedIndex: 0
+    property QtObject selectedSegment
+    property var handleSegmentClick: (() => {});
+
+    onSelectedIndexChanged: {
+        selectedSegment = options.itemAt(selectedIndex)
+    }
+
+    Component.onCompleted: {
+        selectedSegment = options.itemAt(selectedIndex)
+    }
+
+    implicitHeight: VPNTheme.theme.rowHeight
+    color: VPNTheme.theme.input.highlight
+    radius: 24
+
+    Rectangle {
+        id: slider
+        anchors {
+            top: parent.top
+            left: parent.left
+            bottom: parent.bottom
+            topMargin: 4
+            leftMargin: implicitWidth * root.selectedIndex + 4
+            rightMargin: 4
+            bottomMargin: 4
+        }
+
+        implicitWidth: (parent.width - 8) / model.count
+        color: VPNTheme.theme.white
+        radius: parent.radius
+
+        Behavior on anchors.leftMargin {
+            PropertyAnimation {
+                duration: 100
+            }
+
+        }
+    }
+
+    RowLayout {
+
+        anchors {
+            fill: parent
+            topMargin: 8
+            leftMargin: 4
+            rightMargin: 4
+            bottomMargin: 8
+        }
+
+        spacing: 0
+
+        Repeater {
+            id: options
+
+            delegate: VPNButtonBase {
+                id: segment
+
+                Layout.preferredWidth: slider.implicitWidth
+                Layout.fillHeight: true
+
+                objectName: segmentButtonId
+
+                Accessible.description: `${index + 1} ${VPNl18n.AccessiblityOf} ${options.count}`
+                Accessible.name: {
+                    var name = ""
+                    if(root.selectedIndex === index) name += `${VPNl18n.AccessiblitySelected}.`
```

Comment:
If I understand the code (which I'm not sure about), these two strings should be exposed for localization in their entirety, i.e. 
- `%1 of %2`
- `%1 selected`

Hardcoding characters or imposing a specific order (e.g. noun followed by adjective) is generally a [red flag for localization](https://mozilla-l10n.github.io/documentation/localization/dev_best_practices.html#avoid-concatenations-use-placeholders-instead).


I'm still not sure what the referenced noun is (what is selected? 1 of 2 of?), but we should add that to comments in strings.yaml

P.S. I think the linter should have checks for strings.yaml, but there should be an empty line at the end of the file.

Reply:
I have made some changes based on your suggestions, but please let me know if and how it could be improved further!